mlx/ops.cpp

```diff
@@ -1608,6 +1608,34 @@ array logical_not(const array& a, StreamOrDevice s /* = {} */) {
       {astype(a, bool_, s)});
 }
 
+array logical_and(const array& a, const array& b, StreamOrDevice s /* = {} */) {
+  // Ensure both arrays have the same shape
+  if (a.shape() != b.shape()) {
+    throw std::invalid_argument(
+        "Arrays must have the same shape for logical_and");
+  }
```

Comment:
No, they should be allowed to broadcast. See our other binary ops for an example (like `add`).